client/my-sites/hosting/staging-site-card/index.js

```diff
@@ -0,0 +1,121 @@
+import { Button, Card, Spinner } from '@automattic/components';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import { localize } from 'i18n-calypso';
+import { connect, useDispatch, useSelector } from 'react-redux';
+import CardHeading from 'calypso/components/card-heading';
+import MaterialIcon from 'calypso/components/material-icon';
+import { useAddStagingSiteMutation } from 'calypso/my-sites/hosting/staging-site-card/use-add-staging-site';
+import { useCheckStagingSiteStatus } from 'calypso/my-sites/hosting/staging-site-card/use-check-staging-site-status';
+import { useStagingSite } from 'calypso/my-sites/hosting/staging-site-card/use-staging-site';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { transferStates } from 'calypso/state/automated-transfer/constants';
+import { errorNotice, removeNotice, successNotice } from 'calypso/state/notices/actions';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+const noticeOptions = {
+	duration: 3000,
+};
+
+const stagingSiteAddFailureNoticeId = 'staging-site-add-failure';
+
+const StagingSiteCard = ( { disabled, siteId, translate } ) => {
+	const { __ } = useI18n();
+	const dispatch = useDispatch();
+
+	const { data: stagingSites, isLoading: isLoadingStagingSites } = useStagingSite( siteId, {
+		enabled: ! disabled,
+	} );
+
+	const stagingSite = stagingSites && stagingSites.length ? stagingSites[ 0 ] : [];
+	const stagingSiteSlug = useSelector( ( state ) => getSiteSlug( state, stagingSite.id ) );
+
+	const showAddStagingSite = ! isLoadingStagingSites && stagingSites && stagingSites.length === 0;
+	const showManageStagingSite = ! isLoadingStagingSites && stagingSites && stagingSites.length > 0;
+
+	const transferStatus = useCheckStagingSiteStatus( stagingSite.id );
+	const isStagingSiteTransferComplete = transferStatus === transferStates.COMPLETE;
+
+	const { addStagingSite, isLoading: addingStagingSite } = useAddStagingSiteMutation( siteId, {
+		onMutate: () => {
+			dispatch( removeNotice( stagingSiteAddFailureNoticeId ) );
+		},
+		onSuccess: () => {
+			dispatch( recordTracksEvent( 'calypso_hosting_configuration_staging_site_add_success' ) );
+			dispatch( successNotice( __( 'Staging site was added.' ), noticeOptions ) );
```

Comment:
```suggestion
			dispatch( successNotice( __( 'Staging site added.' ), noticeOptions ) );
```

This notice disappeared really quickly for me (much quicker than 3 seconds). Maybe we should make it always present?

Reply:
Actually, now I see we don't need this notice. We display it when the simple site is added, but we still need to wait and poll for the transfer status. What if we remove it completely and solve it under 1767-gh-Automattic/dotcom-forge ?

Reply:
@wojtekn I think a notice would be nice. Can we have it displayed after the staging site is fully created?

Reply:
Moved with 6ef8d1ee1e6e0027a007e18a3e298a9cfd1a41c3